src/libmeasurement_kit/nettests/base_test.cpp

```diff
@@ -98,25 +99,59 @@ BaseTest &BaseTest::on_destroy(Delegate<> cb) {
     return *this;
 }
 
+static void start_internal_(Var<Runnable> &&r, std::promise<void> *promise,
+                            Callback<> &&callback) {
```

Comment:
The core function now used to run tests. The detailed comment should explain all, er, details.